Sources/Datadog/FeaturesIntegration/WebView/WebLogEventConsumer.swift

```diff
@@ -7,7 +7,70 @@
 import Foundation
 
 internal class WebLogEventConsumer: WebEventConsumer {
-    func consume(event: [String: Any], eventType: String) {
-        // TODO: RUMM-1791 implement event consumers
+    private struct Constants {
+        static let logEventType = "log"
+        static let internalLogEventType = "internal_log"
+
+        static let applicationIDKey = "application_id"
+        static let sessionIDKey = "session_id"
+        static let ddTagsKey = "ddtags"
+        static let dateKey = "date"
+    }
+
+    private let userLogsWriter: Writer
+    private let internalLogsWriter: Writer?
+    private let dateCorrector: DateCorrectorType
+    private let rumContextProvider: RUMContextProvider?
+    private let jsonDecoder = JSONDecoder()
+
+    private lazy var ddTags: String = {
+        let versionKey = LogEventEncoder.StaticCodingKeys.applicationVersion.rawValue
+        let versionValue = LoggingFeature.instance?.configuration.common.applicationVersion ?? ""
+        let envKey = LogEventEncoder.StaticCodingKeys.environment.rawValue
+        let envValue = LoggingFeature.instance?.configuration.common.environment ?? ""
+
+        return "\(versionKey):\(versionValue),\(envKey):\(envValue)"
+    }()
+
+    init(
+        userLogsWriter: Writer,
+        internalLogsWriter: Writer?,
+        dateCorrector: DateCorrectorType,
+        rumContextProvider: RUMContextProvider?
+    ) {
+        self.userLogsWriter = userLogsWriter
+        self.internalLogsWriter = internalLogsWriter
+        self.dateCorrector = dateCorrector
+        self.rumContextProvider = rumContextProvider
+    }
+
+    func consume(event: JSON, eventType: String) throws {
+        var mutableEvent = event
+
+        if let existingTags = mutableEvent[Constants.ddTagsKey] as? String, !existingTags.isEmpty {
+            mutableEvent[Constants.ddTagsKey] = "\(ddTags),\(existingTags)"
+        } else {
+            mutableEvent[Constants.ddTagsKey] = ddTags
+        }
+
+        if let date = mutableEvent[Constants.dateKey] as? Int {
+            let serverTimeOffsetInNs = dateCorrector.currentCorrection.serverTimeOffset.toInt64Nanoseconds
+            let correctedDate = Int64(date) + serverTimeOffsetInNs
+            mutableEvent[Constants.dateKey] = correctedDate
+        }
+
+        if let context = rumContextProvider?.context {
+            mutableEvent[Constants.applicationIDKey] = context.rumApplicationID
+            mutableEvent[Constants.sessionIDKey] = context.sessionID.toRUMDataFormat
+        }
+
+        let jsonData = try JSONSerialization.data(withJSONObject: mutableEvent, options: [])
+        let encodableEvent = try jsonDecoder.decode(CodableValue.self, from: jsonData)
+
+        if eventType == Constants.logEventType {
+            userLogsWriter.write(value: encodableEvent)
+        } else if eventType == Constants.internalLogEventType {
+            internalLogsWriter?.write(value: encodableEvent)
+        }
```

Comment:
What if the event is not recognised? We already captured it in the native SDK, but here we resign from processing it so it will be lost. We should at least inform the user on dropping an event and consider adding internal monitoring to spot these cases if they might come from our fault.

Reply:
ℹ️ idk why github doesn't label this line as `outdated` but it is `outdated`

Reply:
```swift
userLogger.error("🔥 Invalid Web Event Type: \(eventType)")
```
How will we know if there is an issue in this system? Say, some Browser SDK version sends unrecognised event type, don't we want to be informed that we drop entire range of events, potentially destroying user's observability in critical flow? I think this case deserves Internal Monitoring observability as well.

Reply:
`WebLogEventConsumer` protocol changed in a way that eliminates this case